flocker/node/agents/cinder.py

```diff
@@ -497,3 +502,52 @@ def cinder_api(cinder_client, nova_client, cluster_id):
         nova_server_manager=logging_nova_server_manager,
         cluster_id=cluster_id,
     )
+
+
+def _openstack_auth_from_config(auth_plugin='password', **config):
+    """
+    Create an OpenStack authentication plugin from the given configuration.
+
+    :param str auth_plugin: The name of the authentication plugin to create.
+    :param config: Parameters to supply to the authentication plugin.  The
+        exact parameters depends on the authentication plugin selected.
+
+    :return: The authentication object.
+    """
+    if auth_plugin == 'rackspace':
+        plugin_class = RackspaceAuth
+    else:
+        plugin_class = get_plugin_class(auth_plugin)
+
+    plugin_options = plugin_class.get_options()
+    plugin_kwargs = {}
+    for option in plugin_options:
+        # option.dest is the python compatible attribute name in the plugin
+        # implementation.
+        # option.dest is option.name with hyphens replaced with underscores.
+        if option.dest in config:
+            plugin_kwargs[option.dest] = config[option.dest]
+
+    return plugin_class(**plugin_kwargs)
+
+
+def cinder_from_configuration(**config):
+    """
+    Build a ``CinderBlockDeviceAPI`` using configuration and
+    credentials in ``config``.
+    """
+    region = config.pop('region')
+    auth = _openstack_auth_from_config(**config)
```

Comment:
Perhaps there should be a sub-section of the config for authentication? Or does it expect things like region too? Seems cleaner to have it in its own namespace in case of conflicts.